frontend/pages/404.tsx

```diff
@@ -0,0 +1,29 @@
+import { Stack, Typography } from '@mui/material';
+import { useRouter } from 'next/router';
+import Button from '../components/generic/Button';
+import RandomKawaii from '../components/notFound/RandomKawaii';
+
+interface Props {
+  entity?: string;
+  message?: string;
+}
+
+function NotFound({ entity = 'page', message }: Props) {
+  const router = useRouter();
+
+  message = message || `The ${entity} you are looking for does not exist.`;
+
+  return (
+    <Stack justifyContent="center" alignItems="center" spacing={2}>
```

Comment:
```suggestion
    <Stack justifyContent="center" alignItems="center" height="100vh" spacing={2}>
```

Reply:
I feel like not centering looks more natural though